src/helpers/validation.ts

```diff
@@ -0,0 +1,21 @@
+import * as yup from "yup";
+
+export const loginSchema = yup.object({
+  username: yup.string().required("Username can't be blank"),
+  password: yup
+    .string()
+    .required("Password can't be blank")
+    .min(8, "Please use minimum of 8 characters for password"),
+});
+
+export const registerSchema = yup.object({
+  username: yup.string().required("Username can't be blank"),
+  password: yup
+    .string()
+    .required("Password can't be blank")
+    .min(8, "Please use minimum of 8 characters for password"),
+  email: yup
+    .string()
+    .required("Email can't be blank")
+    .email("Please use a valid email address"),
+});
```

Comment:
```suggestion
export const registerSchema = loginSchema.concat(
  yup.object({
    email: yup
      .string()
      .required("Email can't be blank")
      .email("Please use a valid email address"),
  })
);
```
you duplicated big part of code.  use the closest thing to schema extension/inheritance in Yup